src/main/scala/dotty/dokka/PackageHierarchyTransformer.scala

```diff
@@ -0,0 +1,62 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.transformers.pages.{PageTransformer}
+import org.jetbrains.dokka.pages._
+import collection.JavaConverters
+import collection.JavaConverters._
+import org.jetbrains.dokka.plugability.DokkaContext
+
+class PackageHierarchyTransformer(context: DokkaContext) extends PageTransformer:
+    override def invoke(input: RootPageNode): RootPageNode = input match {
+        case m: ModulePageNode => rearangePackagePages(m)
+        case other => {
+            context.getLogger.warn("PackageHierarchyTransformer: Couldn't transform root because root is not ModulePageNode")
+            other
+        }
+    }
+
+    def rearangePackagePages(page: ModulePageNode): ModulePageNode = {
+        val packagePages = page.getChildren.asScala.collect{ case p: PackagePageNode => p }.toList
+
+
+        def isParent(possibleParent: Seq[String], comparedChildren: Seq[String]): Boolean = {
+            possibleParent.zip(comparedChildren).forall(_ == _)
```

Comment:
why not `possibleParent == comparedChildren`?

Reply:
possibleParent and comparedChildren are of different sizes, so there would be no equality. Zip ignores remaining element of longer list, so then I can compare elem by elem

Reply:
you can use `startsWith` in such case

Reply:
That's a good point, changed.